cyral/resource_cyral_repository.go

```diff
@@ -114,7 +136,13 @@ func (r *RepoInfo) ReadFromSchema(d *schema.ResourceData) error {
 	r.LabelsFromInterface(d.Get(RepoLabelsKey).([]interface{}))
 	r.RepoNodesFromInterface(d.Get(RepoNodesKey).([]interface{}))
 	r.ConnDrainingFromInterface(d.Get(RepoConnDrainingKey).(*schema.Set).List())
-	r.MongoDBSettingsFromInterface(d.Get(RepoMongoDBSettingsKey).(*schema.Set).List())
+	var mongoDBSettings = d.Get(RepoMongoDBSettingsKey).(*schema.Set).List()
+	if r.Type == MongoDB && (mongoDBSettings == nil || len(mongoDBSettings) == 0) {
+		return fmt.Errorf("'%s' block is mandatory when 'type=%s'", RepoMongoDBSettingsKey, MongoDB)
+	} else if r.Type != MongoDB && len(mongoDBSettings) > 0 {
+		return fmt.Errorf("'%s' block is only allowed when 'type=%s'", RepoMongoDBSettingsKey, MongoDB)
+	}
+	r.MongoDBSettingsFromInterface(mongoDBSettings)
```

Comment:
Within `MongoDBSettingsFromInterface()` there's still a missing check: if the server type is `replicaset`, then the replica set name MUST be supplied. If it's not, then the API will complain, so nothing terrible will happen, but we should probably fail earlier.

Reply:
Perfect. Will add this check. Thanks!